niceman/interface/diff.py

```diff
@@ -0,0 +1,150 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil; coding: utf-8 -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the niceman package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Report if a specification satisfies the requirements in another specification
+"""
+
+import sys
+import time
+
+from .base import Interface
+from ..support.constraints import EnsureStr
+from ..support.exceptions import InsufficientArgumentsError
+from ..support.param import Parameter
+
+__docformat__ = 'restructuredtext'
+
+from logging import getLogger
+lgr = getLogger('niceman.api.retrace')
+
+class MultipleDistributionsError(Exception):
+
+    """multiple distributions of a given type found"""
+
+    def __init__(self, cls):
+        self.cls = cls
+
+class Diff(Interface):
+    """Report if a specification satisfies the requirements in another 
+    specification
+
+    Examples
+    --------
+
+      $ niceman diff --env environment.yml --req required_environment.yml
+
+    """
+
+    _params_ = dict(
+        env=Parameter(
+            args=("--env",),
+            doc="NICEMAN environment specification", 
+            metavar='env',
+            constraints=EnsureStr()),
+        req=Parameter(
+            args=("--req",),
+            metavar="req",
+            doc="NICEMAN requirements", 
+            constraints=EnsureStr())
+    )
```

Comment:
what use-cases do you envision while having those as --arguments?
The simplest use case I see, is very similar to a regular 'diff' command -- `diff [options] spec1 spec2`
Seeing it this way also allows you to simplify code -- consider `specs`  to be a list! 
`__call__(*specs)`
and avoid code duplication you have now via list comprehensions and/or a simple loop.

PS Thinking this way could eventually also lead to another generalization: arbitrary number of specs!  e.g. I collected a bunch of specs in a directory.  Are they different? how?  but that is for later
PSS It seems that my mind is tuned to making everyone do `diff`, so you could see also a work of a new students who decided to help out with nibabel (while learning the "trade"):  `nib-ls diff` https://github.com/yarikoptic/nibabel/pull/3 , where diff'ing multiple niftis is on his agenda as well